cunumeric/module.py

```diff
@@ -5327,6 +5336,630 @@ def nancumsum(
     )
 
 
+def _get_non_nan_reduction_code_if_applicable(
+    dtype: npt.DTypeLike, unary_reduction_code: UnaryRedCode
+) -> UnaryRedCode:
+    """
+    Return the equivalent non-nan reduction op codes if the datatype of the
+    array doesn't belong to one of the allowed datatypes. Raise an error if the
+    datatype is disallowed, which is currently complex64 and complex128.
+    """
+
+    equivalent_non_nan_unary_red_ops: Dict[UnaryRedCode, UnaryRedCode] = {
+        UnaryRedCode.NANARGMAX: UnaryRedCode.ARGMAX,
+        UnaryRedCode.NANARGMIN: UnaryRedCode.ARGMIN,
+        UnaryRedCode.NANMAX: UnaryRedCode.MAX,
+        UnaryRedCode.NANMIN: UnaryRedCode.MIN,
+    }
+
+    assert unary_reduction_code in equivalent_non_nan_unary_red_ops
+
+    # raise error for disallowed datatypes
+    disallowed_dtypes: list[np.dtype[Any]] = [
```

Comment:
same comment here. any constant metadata you want to keep around can be moved to the global scope, so it is created only once. also, consider using tuple instead of list here. list is a mutable data structure, whereas tuple is immutable. so constant metadata like this better be kept in a tuple.